polls/views.py

```diff
@@ -1,36 +1,70 @@
-            # if it's user's first time
-            context = {
-                "should_rest": False,
-                "can_continue": False
-                }
-            return render(request, self.template_name, context)
-
-        current_time = timezone.now()
-        time_diff = current_time - request.user.first_task_of_this_session_performed_at
-        time_diff_minutes = time_diff.total_seconds() / 60
-        should_rest = time_diff_minutes > self.minutes_after_should_rest and \
-                      time_diff_minutes < self.minutes_after_can_continue
-        can_continue = time_diff_minutes > self.minutes_after_can_continue
-        rest_time = round(self.minutes_after_can_continue - time_diff_minutes) + 1
+        can_continue, should_rest, rest_time = check_user_work_permission(request.user)
         context = {
-            "should_rest": should_rest,
             "can_continue": can_continue,
+            "should_rest": should_rest,
             "rest_time": rest_time
         }
-        return render(request, self.template_name, context)
\ No newline at end of file
+        return render(request, self.template_name, context)
+
+class TaskFlowView(LoginRequiredMixin, UserPassesTestMixin, View):
+    template_name = "polls/task_flow.html"
+
+    def get(self, request):
+        can_continue, should_rest, _ = check_user_work_permission(request.user)
+        if should_rest:
+            return redirect("auth-flow")
+        elif can_continue:
+            # Update user's session start time
+            request.user.first_task_of_this_session_performed_at = timezone.now()
+            request.user.save()
+
+        if batch_selector():
+            # when probability lesss than 90%
+            current_batch = CurrentBatchEval.objects.first().current_batch_eval
+        else:
+            current_batch = CurrentBatchGold.objects.first().current_batch_gold
+        all_tasks = current_batch.tasks.all()
+        tasks_for_user = all_tasks.exclude(annotation__user=request.user)
+        task = tasks_for_user.first()
+        context = { "task": task }
+        if not task:
+            return render(request, self.template_name, context)
+        url, task_presentation = present_task_for_user(task)
+        context["url"] = url
+        context["task_presentation"] = task_presentation      
+        return render(request, self.template_name, context)
+
+    def post(self, request):
+        annotation_choice = request.POST.get("annotationOption")
+        task_pk = request.POST.get("taskPk")
+        task_presentation = request.POST.get("taskPresentation")
+        Annotation.objects.create(
+            user=request.user,
+            task=Task.objects.get(pk=task_pk),
+            annotated_at=timezone.now(),
+            task_presentation=task_presentation,
+            annotations=annotation_choice
+        )
+        return redirect("task-flow")
+
+    def test_func(self):
```

Comment:
This is a really generic name. Please rename to something better and more self-explanatory.

Reply:
This is required by `UserPassesTestMixin` class, renaming will cause it malfunction

Reply:
Ok, I will implement it this way

Reply:
@Programmer-yar so link to the issue you created??

Reply:
https://github.com/turian/audio-discrimination-crowdsource/issues/20